classloader-leak-prevention/classloader-leak-prevention-core/src/main/java/se/jiderhamn/classloader/leak/prevention/cleanup/StopThreadsCleanUp.java

```diff
@@ -222,12 +222,25 @@ else if(preventor.isLoadedInClassLoader(executor) || preventor.isLoadedInClassLo
                     " ms. " + preventor.getStackTrace(thread));
                 preventor.waitForThread(thread, threadWaitMs, false /* No interrupt */);
               }
-              
+
               if(thread.isAlive() && preventor.isClassLoaderOrChild(thread.getContextClassLoader())) { // Still running in ClassLoader
                 preventor.warn(displayString + (waitForThreads ? " still" : "") + 
                     " alive; changing context ClassLoader to leak safe (" + 
                     preventor.getLeakSafeClassLoader() + "). " + preventor.getStackTrace(thread));
                 thread.setContextClassLoader(preventor.getLeakSafeClassLoader());
+
+                // Replace AccessControlContext since we already replaced ClassLoader, @see StopThreadsClenup_ExecutorTest
```

Comment:
Seems it would make sense to share this code with line 157-. Would you care to extract a method, @skarzhevskyy ?

Reply:
@mjiderhamn  No problem